crates/cxx-qt-lib-extras/src/assertion_utils.h

```diff
@@ -14,27 +15,39 @@
 namespace rust::cxxqtlib1::assertion_utils {
 template<typename iter>
 constexpr static ::std::size_t
-calc_align_size(const iter rbegin,
-                const iter rend,
+calc_align_size(const iter begin,
+                const iter end,
                 const ::std::size_t actual_alignment)
 {
   ::std::size_t rows = 0;
-  ::std::size_t accum = 0;
 
-  for (auto it = rbegin; it != rend; ++it) {
-    assert(*it <= actual_alignment);
+  ::std::size_t row_accum = 0;
+  ::std::size_t max_size_in_row = 0;
 
-    if (it + 1 != rend) {
-      if (accum + *it == actual_alignment) {
-        accum += *it;
+  for (auto it = begin; it != end; ++it) {
+    auto val = *it;
+    assert(val <= actual_alignment);
+
+    if (val > max_size_in_row)
+      max_size_in_row = val;
+
+    if ((row_accum + val) % max_size_in_row == 0) {
```

Comment:
I believe this is incorrect.
Currently, it checks whether the alignment of the new value is correctly aligned with whatever the largest value in this row was.
But it should really check whether the alignment of the new value is correctly aligned with its own alignment rules.

Changing this to: `(row_accum + val) % max_size_in_row == 0` should fix it for all trivial types, as those always have `sizeof() == alignof()`.
Though in reality we would really like to check whether `val` is `% alignof(the type val represents)`.